packages/nest/src/schematics/library/library.ts

```diff
@@ -0,0 +1,186 @@
+import { normalize, Path } from '@angular-devkit/core';
+import {
+  apply,
+  chain,
+  externalSchematic,
+  filter,
+  MergeStrategy,
+  mergeWith,
+  move,
+  noop,
+  Rule,
+  SchematicContext,
+  template,
+  Tree,
+  url
+} from '@angular-devkit/schematics';
+import {
+  addGlobal,
+  formatFiles,
+  getNpmScope,
+  getProjectConfig,
+  insert,
+  names,
+  offsetFromRoot,
+  toFileName,
+  updateJsonInTree,
+  updateWorkspaceInTree
+} from '@nrwl/workspace';
+import { Schema } from './schema';
+import * as ts from 'typescript';
+import { RemoveChange } from '@nrwl/workspace/src/utils/ast-utils';
+
+export interface NormalizedSchema extends Schema {
+  name: string;
+  prefix: string;
+  fileName: string;
+  projectRoot: Path;
+  projectDirectory: string;
+  parsedTags: string[];
+}
+
+export default function(schema: NormalizedSchema): Rule {
+  return (host: Tree, context: SchematicContext) => {
+    const options = normalizeOptions(host, schema);
+
+    return chain([
+      externalSchematic('@nrwl/node', 'lib', schema),
+      createFiles(options),
+      addExportsToBarrelFile(options),
+      updateTsConfig(options),
+      addProject(options),
+      formatFiles(options)
+    ]);
+  };
+}
+
+function normalizeOptions(host: Tree, options: Schema): NormalizedSchema {
+  const defaultPrefix = getNpmScope(host);
+  const name = toFileName(options.name);
+  const projectDirectory = options.directory
+    ? `${toFileName(options.directory)}/${name}`
+    : name;
+
+  const projectName = projectDirectory.replace(new RegExp('/', 'g'), '-');
+  const fileName = projectName;
+  const projectRoot = normalize(`libs/${projectDirectory}`);
+
+  const parsedTags = options.tags
+    ? options.tags.split(',').map(s => s.trim())
+    : [];
+
+  const normalized: NormalizedSchema = {
+    ...options,
+    prefix: defaultPrefix, // we could also allow customizing this
+    fileName,
+    name: projectName,
+    projectRoot,
+    projectDirectory,
+    parsedTags
+  };
+
+  return normalized;
+}
+
+function addExportsToBarrelFile(options: NormalizedSchema): Rule {
+  return (host: Tree) => {
+    const indexFilePath = `${options.projectRoot}/src/index.ts`;
+    const buffer = host.read(indexFilePath);
+    if (!!buffer) {
+      const indexSource = buffer!.toString('utf-8');
+      const indexSourceFile = ts.createSourceFile(
+        indexFilePath,
+        indexSource,
+        ts.ScriptTarget.Latest,
+        true
+      );
+
+      insert(host, indexFilePath, [
+        new RemoveChange(indexFilePath, 0, "export * from './lib/my-lib';"),
```

Comment:
This is what's causing the lint e2e's to fail. The name generated doesn't match `my-lib` (since that's hardcoded) and doesn't remove the old export. 
```suggestion
        new RemoveChange(indexFilePath, 0, `export * from './lib/{options.fileName}';`),
```

Reply:
Ah yes. Good catch. That's what I get for working on this at night. Thanks!